### DISCUSSION TITLE
Adding an external crystal to the PCB

### DISCUSSION THREAD

**Discussion by EDsteve** [LIFsCode/ELOC-3.0#90] [General]
As already discussed [here](https://github.com/LIFsCode/ELOC-3.0/issues/77#top) and here https://github.com/LIFsCode/ELOC-3.0/issues/11 a external quarz would solve the time-issue and makes BLE possible. 

### Problem
The crystal needs GPIO33 and 32 on the ESP32 which is occupied by MISO_LoRa and SCK_LoRa. Unfortunately we don't have enough GPIO pins available any more to move MISO_LoRa and SCK_LoRa within the ESP32. 

### Solution
Usinge the IO expander. These pins should be not too critical and could be moved to the IO expander? Please correct me if i am wrong:
- Reset_LoRa on pin 25
- TXEN_LoRa on pin 4 (Since we are using DI01 for LoRa, this pin is not necessary any more and basically free to use?)
- BUSY_LoRa on pin 35
- Accel_Trigger on pin 12

Moving the LoRa pins to the IO expander would have the least impact on the ELOC PCB because the LoRa module is connected directly to the IO expander via pin headers.

So i would suggest the following:

- Reset_LoRa (before GPIO 25) -> IO expander GPIO 5
- MISO_LoRa (before GPIO 32) -> GPIO 25
- Remove TXEN_LoRa (before GPIO 04)
- SCK_LoRa (before GPIO 33) -> GPIO 04
- Crystal -> GPIO 33 and GPIO 32

An overview of the ELOC pinout is available [here](https://docs.google.com/spreadsheets/d/1upqkyV_vz4PRbfbt1Fz_Z0wQGxAQJCGD2hbBw_J68x4/edit?usp=sharing)

**Comment by LIFsCode** [comment]
@EDsteve your pin out suggestions look good, but I just had a quick look. 👍

A few comments on the signals:

- Accel trigger **must** be connected to the ESP32 if it should be used to wake up the device from (deep) sleep. This was requested earlier if I remember correctly.
- Lora Dio1 must be connected to ESP the Lora libraries use this as interrupt source and it's timing critical.
- Same may apply for Busy but here I'm not sure I have to look it up
- reset should not be timing critical so I guess it could be safely moved to IO expander. But I'll check that
- Lora TXen and RXen are usually handled by the Lora module itself and the external signals are only to overwrite it. But this must be checked with potential Lora modules

**Comment by OOHehir** [comment]
@EDsteve @LIFsCode Just had a quick look, but this is suggestion is probably not possible. 

The ability to debug using JTAG would be a useful feature, the (fixed) required pins are listed [here](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/jtag-debugging/tips-and-quirks.html#can-jtag-pins-be-used-for-other-purposes).  I had thought that it might be possible to have a method (cutting traces with a header etc) of disconnecting not critical hardware (buzzer etc) during debugging .The main pin issues are GPIO 14 & 15 (used by SD card). It seems these can't be changed.

**Reply by EDsteve** [reply]
Not sure how to solve that problem. Or you have something in mind?
